File: newsletter/generate.py
#!/usr/bin/env python3
"""Newsletter Generator - Creates daily HTML newsletter"""

import os
import json
import requests
from datetime import datetime

API_URL = "http://localhost:3001"
OUTPUT_DIR = "/root/.openclaw/workspace/newsletter"
DATE = datetime.now().strftime("%Y-%m-%d")

os.makedirs(OUTPUT_DIR, exist_ok=True)

def get_content():
    """Get content stats from API"""
    try:
        r = requests.get(f"{API_URL}/api/content-simple", timeout=5)
        data = r.json()
        tiktok = data.get('tiktok', [])
        youtube = data.get('youtube', [])
        
        # Top content
        all_content = tiktok + youtube
        sorted_content = sorted(all_content, key=lambda x: x.get('views', 0), reverse=True)
        
        return {
            'tiktok_count': len(tiktok),
            'youtube_count': len(youtube),
            'total_views': sum(v.get('views', 0) for v in all_content),
            'top_content': sorted_content[:5]
        }
    except Exception as e:
        print(f"Error getting content: {e}")
        return {'tiktok_count': 0, 'youtube_count': 0, 'total_views': 0, 'top_content': []}

def get_research():
    """Get recent research"""
    try:
        r = requests.get(f"{API_URL}/api/investigaciones", timeout=5)
        return r.json()[:5] if r.status_code == 200 else []
    except:
        return []

# Get data
print(f"📰 Generating newsletter for {DATE}...")
content = get_content()
research = get_research()

# Build top content HTML
top_html = ""
for c in content['top_content']:
    title = c.get('titulo', 'Sin título')[:60]
    views = c.get('views', 0)
    platform = '🎵' if 'tiktok' in c.get('video_url', '') else '📺'
    top_html += f"<li>{platform} {title}... ({views:,} views)</li>\n"

# Build research HTML
research_html = ""
for r in research:
    tema = r.get('tema', 'Sin tema')[:80]
    research_html += f"<li>🔬 {tema}</li>\n"

if not research_html:
    research_html = "<li>No hay investigaciones recientes</li>"

# Generate HTML
html = f"""<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>OpenClaw Newsletter - {DATE}</title>
  <style>
    body {{ font-family: 'Segoe UI', sans-serif; max-width: 600px; margin: 0 auto; padding: 20px; background: #0f172a; color: white; }}
    h1 {{ color: #22c55e; border-bottom: 2px solid #22c55e; padding-bottom: 10px; }}
    h2 {{ color: #8b5cf6; margin-top: 30px; }}
    .stats {{ display: flex; gap: 20px; margin: 20px 0; }}
    .stat {{ background: #1e293b; padding: 15px 25px; border-radius: 10px; text-align: center; }}
    .stat-value {{ font-size: 1.5rem; font-weight: bold; color: #22c55e; }}
    .stat-label {{ font-size: 0.85rem; color: #64748b; }}
    ul {{ line-height: 1.8; }}
    li {{ margin: 8px 0; }}
    .footer {{ margin-top: 40px; padding-top: 20px; border-top: 1px solid #334155; color: #64748b; font-size: 0.85rem; }}
    .highlight {{ background: linear-gradient(90deg, #22c55e, #8b5cf6); -webkit-background-clip: text; -webkit-text-fill-color: transparent; }}
    a {{ color: #22c55e; }}
  </style>
</head>
<body>
  <h1>📰 OpenClaw Daily Digest</h1>
  <p><strong>Fecha:</strong> {DATE}</p>
  
  <h2>📊 Estado del Sistema</h2>
  <div class="stats">
    <div class="stat">
      <div class="stat-value">{content['tiktok_count']}</div>
      <div class="stat-label">TikTok Videos</div>
    </div>
    <div class="stat">
      <div class="stat-value">{content['youtube_count']}</div>
      <div class="stat-label">YouTube Videos</div>
    </div>
    <div class="stat">
      <div class="stat-value">{content['total_views']:,}</div>
      <div class="stat-label">Total Views</div>
    </div>
  </div>

  <h2>🔬 Investigaciones Recientes</h2>
  <ul>
    {research_html}
  </ul>

  <h2>🔥 Top Contenido</h2>
  <ul>
    {top_html}
  </ul>

  <div class="footer">
    <p>🤖 Generado automáticamente por <span class="highlight">OpenClaw</span></p>
    <p>📊 <a href="http://212.227.107.120:4000/">Ver Dashboard</a></p>
  </div>
</body>
</html>"""

# Save
output_file = f"{OUTPUT_DIR}/newsletter-{DATE}.html"
with open(output_file, 'w') as f:
    f.write(html)

print(f"✅ Newsletter saved: {output_file}")
print(f"   TikTok: {content['tiktok_count']}")
print(f"   YouTube: {content['youtube_count']}")
print(f"   Total Views: {content['total_views']:,}")

File: presentation/generate.js
/**
 * Presentation Generator
 * Creates beautiful HTML presentations from research data
 */

const fs = require('fs');
const path = require('path');

// Generate a beautiful presentation
function generatePresentation(data) {
  const { title, subtitle, sections, charts } = data;
  
  let slides = '';
  
  // Title slide
  slides += `
  <div class="slide title-slide active">
    <div class="title-content">
      <h1>${title || 'Presentación'}</h1>
      <p class="subtitle">${subtitle || ''}</p>
      <p class="date">${new Date().toLocaleDateString('es-ES', { year: 'numeric', month: 'long', day: 'numeric' })}</p>
    </div>
  </div>`;
  
  // Content slides
  if (sections) {
    sections.forEach((section, i) => {
      slides += `
  <div class="slide">
    <h2>${section.title}</h2>
    <div class="content">
      ${section.content}
    </div>
  </div>`;
    });
  }
  
  // Chart slides
  if (charts) {
    charts.forEach(chart => {
      slides += `
  <div class="slide chart-slide">
    <h2>${chart.title}</h2>
    <div class="chart-container">
      <canvas id="chart-${chart.id}"></canvas>
    </div>
  </div>`;
    });
  }
  
  // Final slide
  slides += `
  <div class="slide end-slide">
    <h1>Gracias</h1>
    <p>🤖 Generado por OpenClaw</p>
  </div>`;
  
  return slides;
}

// Full HTML template
function createPresentation(data) {
  const slides = generatePresentation(data);
  
  return `<!DOCTYPE html>
<html lang="es">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>${data.title || 'Presentación'}</title>
  <script src="https://cdn.jsdelivr.net/npm/chart.js"><\/script>
  <style>
    * { margin: 0; padding: 0; box-sizing: border-box; }
    body { 
      font-family: 'Segoe UI', system-ui, sans-serif; 
      background: #0f172a; 
      color: white;
      overflow: hidden;
    }
    .slide {
      display: none;
      height: 100vh;
      padding: 60px;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      background: linear-gradient(135deg, #0f172a 0%, #1e293b 100%);
    }
    .slide.active { display: flex; }
    
    .title-slide {
      text-align: center;
      background: linear-gradient(135deg, #1e1e3f 0%, #0f172a 100%);
    }
    .title-content h1 {
      font-size: 4rem;
      background: linear-gradient(90deg, #8b5cf6, #22c55e);
      -webkit-background-clip: text;
      -webkit-text-fill-color: transparent;
      margin-bottom: 20px;
    }
    .subtitle { font-size: 1.5rem; color: #94a3b8; }
    .date { font-size: 1rem; color: #64748b; margin-top: 30px; }
    
    h2 {
      font-size: 2.5rem;
      color: #8b5cf6;
      margin-bottom: 40px;
      text-align: center;
    }
    
    .content {
      font-size: 1.3rem;
      line-height: 2;
      max-width: 900px;
      text-align: center;
    }
    
    .content ul { text-align: left; }
    .content li { margin: 15px 0; }
    
    .chart-container {
      width: 80%;
      max-width: 800px;
      height: 400px;
    }
    
    .end-slide { text-align: center; }
    .end-slide h1 { font-size: 3rem; color: #22c55e; }
    .end-slide p { font-size: 1.2rem; color: #64748b; margin-top: 20px; }
    
    .nav {
      position: fixed;
      bottom: 30px;
      left: 50%;
      transform: translateX(-50%);
      display: flex;
      gap: 20px;
      z-index: 100;
    }
    .nav button {
      background: #334155;
      color: white;
      border: none;
      padding: 15px 30px;
      border-radius: 10px;
      cursor: pointer;
      font-size: 1rem;
    }
    .nav button:hover { background: #8b5cf6; }
    
    .progress {
      position: fixed;
      bottom: 0;
      left: 0;
      height: 4px;
      background: linear-gradient(90deg, #8b5cf6, #22c55e);
      transition: width 0.3s;
    }
    
    .slide-number {
      position: fixed;
      bottom: 30px;
      right: 30px;
      color: #64748b;
      font-size: 0.9rem;
    }
  </style>
</head>
<body>
  ${slides}
  
  <div class="progress" id="progress"></div>
  <div class="slide-number" id="slide-number">1 / 1</div>
  
  <div class="nav">
    <button onclick="prev()">◀ Anterior</button>
    <button onclick="next()">Siguiente ▶</button>
  </div>

  <script>
    let current = 0;
    const slides = document.querySelectorAll('.slide');
    const total = slides.length;
    
    function show() {
      slides.forEach((s, i) => s.classList.toggle('active', i === current));
      document.getElementById('progress').style.width = ((current + 1) / total * 100) + '%';
      document.getElementById('slide-number').innerText = (current + 1) + ' / ' + total;
    }
    
    function next() { current = (current + 1) % total; show(); }
    function prev() { current = (current - 1 + total) % total; show(); }
    
    document.addEventListener('keydown', e => {
      if(e.key === 'ArrowRight') next();
      if(e.key === 'ArrowLeft') prev();
    });
    
    // Render charts if any
    const charts = ${JSON.stringify(data.charts || [])};
    charts.forEach(chart => {
      if (window.Chart) {
        const ctx = document.getElementById('chart-' + chart.id);
        if (ctx) {
          new Chart(ctx, {
            type: chart.type || 'bar',
            data: chart.data,
            options: {
              responsive: true,
              plugins: { legend: { labels: { color: 'white' } } },
              scales: {
                x: { ticks: { color: '#94a3b8' }, grid: { color: '#334155' } },
                y: { ticks: { color: '#94a3b8' }, grid: { color: '#334155' } }
              }
            }
          });
        }
      }
    });
    
    show();
  </script>
</body>
</html>`;
}

// Save presentation
function savePresentation(data, filename) {
  const html = createPresentation(data);
  const filepath = `/root/.openclaw/workspace/ai-media-app/public/presentations/${filename}`;
  fs.writeFileSync(filepath, html);
  return filepath;
}

// CLI
const args = process.argv.slice(2);
if (args[0] === 'generate') {
  // Example usage
  const exampleData = {
    title: 'Resumen de Investigación',
    subtitle: 'IA y Machine Learning 2026',
    sections: [
      { title: 'Introducción', content: '<p>Resumen de las últimas tendencias en IA...</p>' },
      { title: 'Hallazgos Principales', content: '<ul><li>GPT-5 tiene 10x más parámetros</li><li>Claude mejora en coding</li><li>Agents son el nuevo trend</li></ul>' }
    ]
  };
  
  const filepath = savePresentation(exampleData, 'research-demo.html');
  console.log('Created:', filepath);
}

module.exports = { createPresentation, savePresentation };
